osu.Game/Screens/Select/Leaderboards/BeatmapLeaderboard.cs

```diff
@@ -182,9 +182,15 @@ protected override APIRequest FetchScores(Action<IEnumerable<ScoreInfo>> scoresC
             return req;
         }
 
-        protected override LeaderboardScore CreateDrawableScore(ScoreInfo model, int index) => new LeaderboardScore(model, index, IsOnlineScope)
+        protected override LeaderboardScore CreateDrawableScore(ScoreInfo model, int index)
         {
-            Action = () => ScoreSelected?.Invoke(model)
-        };
+            model.Beatmap = beatmap;
+
+            return new LeaderboardScore(model, index, IsOnlineScope)
+            {
+                Action = () => ScoreSelected?.Invoke(model),
+                RefreshAction = () => this.RefreshScores()
```

Comment:
this should not be a thing. subscribe to the `ScoreManager` for changes.

Reply:
I am a bit confused about what you meant by subscribing to `ScoreManager`. I did this in attempt to have the ability to refresh the score board once the local score is deleted. This is based on `BeatmapDetailArea.Leaderboard.RefreshScores()`. This is dependent on the learderboard and not the `ScoreManager`.

Do you have any suggestion on how to refresh the score board inside `LeaderboardScore`? Any advise would be appreciated. Unless we don't think that is a necessary feature.

Reply:
`ScoreManager`, as all `ArchiveModelManager`s, exposes an event that fires whenever an item is deleted:
https://github.com/ppy/osu/blob/d63ab362dcb94f4e56f84853a2b36c0bbb0c8c93/osu.Game/Database/ArchiveModelManager.cs#L61-L65
I'm reasonably confident that's what's being proposed here. Mind the xmldoc though (the note about the callback not being called on the update thread) - you'll have to schedule the refresh. Have a look around the repository for other usages of `Item{Added,Removed}` for an example how to do so.

Reply:
Precisely. Should be subscribing to both in the local leaderboard and updating display.

Reply:
Thanks for the information. This made a lot more sense than whatever I was doing.

I have updated it and subscribed to the `scoreManager` from `BeatmapLeaderboard`  which would affect both the local leaderboard and the beatmap detail area (I am assuming that is what you are referring to for updating display).

After making the changes, the score board for both beatmap detail are and local leaderboard were able to refresh after deleting a score. However, the only issue is that the refresh screen will also get triggered when doing clearing all local score for every local score.

This might be an issue if there are a lot of local scores.